### PR TITLE
Feature/menus All basic menus (without using Managers)

### PR DESCRIPTION
**Pull request by jacqu3sk**
All the menus work but they don't actually do anything yet I will make them do stuff as soon as managers are implemented.

### CODE REVIEW

**Review by RaiderRoss — APPROVED** [review]

File: src/menus/base/IMenu.h
Comment (RKruse42) [inline src/menus/base/IMenu.h:1]: #include <algorithm>
#include <limits>

File: src/menus/base/IMenu.h [hunk 1]
```diff
@@ -0,0 +1,194 @@
+#ifndef IMENU_H
+#define IMENU_H
+
+#include <string>
+#include <vector>
+#include <iostream>
+
```
Comment (RKruse42) [inline src/menus/base/IMenu.h:7]: please include algorithm and limits for non-macOS users

**Review by RKruse42 — CHANGES_REQUESTED** [review]
please include algorithm and limits for non-macOS users in IMenu.h

**Review by 0ptim0s — COMMENTED** [review]
ye

File: src/city/City.h [hunk 2]
```diff
@@ -22,14 +22,23 @@ class City
...
-    static City &instance();
+    /**
+     * @brief Returns a pointer to the singleton instance of City.
+     * @return A pointer to the City instance.
+     */
+    static City *instance();
 
+    // Deleted copy constructor and assignment operator to enforce singleton pattern
     City(const City &) = delete;
     City &operator=(const City &) = delete;
```
Comment (CartographySilence) [inline src/city/City.h:38]: erm what the sigma

**Review by 0ptim0s — APPROVED** [review]
ye

**Review by CartographySilence — APPROVED** [review]
Ok buddy ill let you off this time

**Pull request merged**